### ISSUE TITLE
IE11 issue with .onDrop() method

### ISSUE THREAD

**Issue by tymochko** [okonet/react-dropzone#358]
I am using separate button to initiate upload files (besides Dropzone).
So as described in this library `onClick()` event on button is calling `this.refs.dropzone.open();`

But in IE11 it works as expected only after first click. After second click, method .onDrop() is called before .open().

Also, if user drag'n'drops folder into Dropzone - File is created with `size: 0` in all browsers except IE11, where it is not created at all. It's a pity because we could handle this situation to warn user that folders shouldn't be dropped.
My suggestion is to make IE11 return the same object as in other browsers or at least one property `size: 0`

**Comment by okonet** [comment]
You're welcome to create a PR for that!

**Comment by tswaters** [comment]
There seems to be some timing problems here with Edge as well.

We have `onClick` that basically does `setState({error: null})` and  `onDropRejected` that `setState({error: 'some string'})` -- we're seeing the `onClick` fire multiple times, even after onDropRejected -- so the error never causes a re-render with the message.

I'll see if I can't put together a PR for this.

**Comment by tswaters** [comment]
My mistake, after reading this a bit closer I've got a separate issue, I'll create a different ticket for that.

**Comment by okonet** [comment]
This should have been fixed in #450

**Comment by tymochko** [comment]
Order of method called works fine now.
But can still reproduce this one

> Also, if user drag'n'drops folder into Dropzone - File is created with size: 0 in all browsers except IE11, where it is not created at all.

**Comment by okonet** [comment]
Is the second part of this issue a separate issue? If so, please create a separate issue for it with a detailed description and steps to reproduce. At the moment, I don't have enough capability to work on this lib so if you can fix it, please go ahead.

**Comment by tswaters** [comment]
FWIW, I think dropping of folders will not be possible to fix.  

Under current APIS, the inconsistency results from, I think, the  [DND spec](https://www.w3.org/TR/2010/WD-html5-20101019/dnd.html) being ambiguous on how to deal with directories.... all it says is "files" -- ctrl-F on that document for `directory` returns 0 results.  

When dragging a directory most browsers report a single file (or multiple, if multiple directories selected), each "file" has a size of 0.  In IE11, it returns an array of 0 for files.

Ehh, having said that there are newer apis we might be able to hook into to get at directories (FileSystemAPI), but the support is not very good: http://caniuse.com/filesystem/embed
